packages/components/src/DataList/components/DataListBulkActions/DataListBulkActions.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { Children, ReactElement, isValidElement } from "react";
+import classNames from "classnames";
+import { useInView } from "@jobber/hooks/useInView";
+import { Tooltip } from "@jobber/components/Tooltip";
+import { Button } from "@jobber/components/Button";
+import styles from "./DataListBulkActions.css";
+import { BULK_ACTIONS_CONTAINER_TEST_ID } from "./DataListBulkActions.const";
+import { DataListBulkActionsProps } from "../../DataList.types";
+import { useDataListContext } from "../../context/DataListContext";
+import { DataListItemActionsOverflow } from "../DataListItemActions/DataListItemActionsOverflow";
+import { getExposedActions } from "../../DataList.utils";
+import { useResponsiveSizing } from "../../hooks/useResponsiveSizing";
+
+// This component is meant to capture the props of the DataList.BulkActions
+export function DataListBulkActions(
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  _: DataListBulkActionsProps,
+) {
+  return null;
+}
+
+export function InternalDataListBulkActions() {
+  const { bulkActionsComponent } = useDataListContext();
+  if (!bulkActionsComponent) return null;
+
+  const { children } = bulkActionsComponent.props;
+  const childrenArray =
+    Children.toArray(children).filter<ReactElement>(isValidElement);
+
+  const { sm } = useResponsiveSizing();
+
+  // Collapse all actions under "More actions" when breakpoint is smaller than sm
+  const exposedActions = sm ? getExposedActions(childrenArray, 3) : [];
+  childrenArray.splice(0, exposedActions.length);
+
+  const [leftRef, isLeftVisible] = useInView<HTMLSpanElement>();
+  const [rightRef, isRightVisible] = useInView<HTMLSpanElement>();
```

Comment:
we should extract this overflowing part out into its own component and use here and on the filters instead of duplicating it